README.md

```diff
@@ -108,43 +108,94 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
+Serialization is the process of converting an object into a stream of bytes to store the object in memory, a file or database, or transmit it through network. Its main purpose is to save the state of an object in order to be able to recreate it when needed. The reverse process is called deserialization. Hazelcast offers you its own native serialization types. You will see these types throughout the chapter.
 
-For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic between the client and the server and the performance would severely drop.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+Hazelcast serializes all your objects before sending them to the server. For the following types Hazelcast optimizes the serialization and you cannot override this behavior.
 
-The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
+- A `boolean` type which is a primitive on the Node.js client can be serialized as a `Boolean` for Java server side.
```

Comment:
I think Java counterparts are not necessary at this point. We can just give a list of the types here. Then, we can add a table below to show which type corresponds to which data type in Java.